app/api/logging/flask_logger.py

```diff
@@ -47,13 +49,28 @@ def init_app(app_logger: logging.Logger, app: flask.Flask) -> None:
         handler.addFilter(_add_app_context_info_to_log_record)
         handler.addFilter(_add_request_context_info_to_log_record)
 
+    # Add request context data to every log record for the current request
+    # such as request id, request method, request path, and the matching Flask request url rule
+    app.before_request(
+        lambda: add_extra_data_to_current_request_logs(_get_request_context_info(flask.request))
+    )
+
     # Use the app_logger to log every non-404 request before each request
     # See https://flask.palletsprojects.com/en/2.2.x/api/#flask.Flask.before_request
     app.before_request(lambda: _log_route(app_logger))
 
     app_logger.info("initialized flask logger")
 
 
+def add_extra_data_to_current_request_logs(data: dict[str, str]) -> None:
```

Comment:
```suggestion
def add_extra_data_to_current_request_logs(data: dict[str, str | int | float]) -> None:
```
not every project will necessarily use the same tool for metrics, but allowing additional types (or just doing Any and not worrying) might be better. 

A quick check for NewRelic, I know str/int/float all work - I'd assume these should all be acceptable.

Reply:
hmm good point. i actually think we'll want to support any JSON primitive including bool and None/null type as well. i added your change and added boolean + None too